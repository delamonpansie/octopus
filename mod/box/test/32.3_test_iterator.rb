#!/usr/bin/ruby

$: << File.dirname($0) + '/lib'
require 'run_env'

class Env < RunEnv
  def config
    super + <<EOD
object_space[0].enabled = 1
object_space[0].index[0].type = "TREE"
object_space[0].index[0].unique = 1
object_space[0].index[0].key_field[0].fieldno = 0
object_space[0].index[0].key_field[0].type = "STR"

EOD
  end
end

Env.connect_eval do
  keys = %w{00 aa abc abcd abce def}
  keys.each do |i| insert_nolog [i] end

  log_try { lua 'user_proc.iterator3' }
end
