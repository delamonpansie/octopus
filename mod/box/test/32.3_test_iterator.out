# box.lua("user_proc.iterator3")
Failed with: {code: 0x202, message: 'mod/box/src-lua/box/example_proc.lua:255: context switch during index iteration
stack traceback:
	[C]: in function 'error'
	src-lua/index.lua:203: in function '(for generator)'
	mod/box/src-lua/box/example_proc.lua:255: in function 'proc'
	mod/box/src-lua/box.lua:389: in function <mod/box/src-lua/box.lua:381>'}
